pig/src/main/java/org/jboss/pnc/bacon/pig/impl/repo/RepoManager.java

```diff
@@ -334,6 +371,178 @@ public RepositoryData generate() {
+                    extensionArtifacts.forEach((extensionArtifact -> {
+                        GAV gav = artifactWrapper.toGAV();
+                        if (extensionArtifact.getGroupId().equals(gav.getGroupId())
+                                && extensionArtifact.getArtifactId().equals(gav.getArtifactId())) {
+                            result.put(extensionArtifact, gav.getVersion());
+                        }
+                    }));
+                }));
+            }
+            if (pncBuild.getDependencyArtifacts() != null) {
+                pncBuild.getDependencyArtifacts().forEach((artifactWrapper -> {
+                    extensionArtifacts.forEach((extensionArtifact -> {
+                        GAV gav = artifactWrapper.toGAV();
+                        if (extensionArtifact.getGroupId().equals(gav.getGroupId())
+                                && extensionArtifact.getArtifactId().equals(gav.getArtifactId())) {
+                            result.put(extensionArtifact, gav.getVersion());
+                        }
+                    }));
+                }));
+            }
+        });
+
+        return result;
+    }
+
+    /**
+     * @param urlString url to extensionList file. File must contains the extensions in format of
+     *        groupId:artifactId:version:type:classifier classifier and type are optionals
+     * @return List of Artifacts
+     * @throws Exception
+     */
+    public List<Artifact> parseExtensionsArtifactList(String urlString) throws Exception {
+        try {
+
+            URL url = new URL(urlString);
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            BufferedReader br = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+            ArrayList<Artifact> list = new ArrayList<>();
+            String buffer = "";
+            while (buffer != null) {
+                buffer = br.readLine();
+                if (buffer == null) {
+                    break;
+                }
+                String[] parts = buffer.split(":");
+                if (parts.length < 2 || parts.length > 5) {
+                    throw new Exception("Extension text file is not properly formatted");
+                }
+
+                Artifact extensionRtArtifact;
+                if (parts.length == 3) {
+                    extensionRtArtifact = new DefaultArtifact(parts[0], parts[1], null, "jar", parts[2]);
+                } else if (parts.length == 4) {
+                    extensionRtArtifact = new DefaultArtifact(parts[0], parts[1], null, parts[3], parts[2]);
+                } else {
+                    extensionRtArtifact = new DefaultArtifact(parts[0], parts[1], parts[4], parts[3], parts[2]);
+                }
+                list.add(extensionRtArtifact);
+            }
+            return list;
+        } catch (MalformedURLException mfe) {
+            throw new MalformedURLException("url is not proper " + urlString);
```

Comment:
I think it would be better to do: `throw new RuntimeException("url is not proper " + urlString, mfe)`

Reply:
What about instead using Bacon native exceptions e.g.
```
throw new FatalException("Malformed URL {}", urlString, mfe)
```
?

Reply:
ok using FatalException